src/citrine/resources/branch.py

```diff
@@ -0,0 +1,100 @@
+from typing import Optional, Union, Iterator
+from uuid import UUID
+
+from citrine._rest.collection import Collection
+from citrine._rest.resource import Resource, ResourceTypeEnum
+from citrine._serialization import properties
+from citrine._session import Session
+from citrine._utils.functions import format_escaped_url
+from citrine.resources.design_workflow import DesignWorkflowCollection
+from citrine.resources.response import Response
+
+
+class Branch(Resource['Branch']):
+    _response_key = None
+    _resource_type = ResourceTypeEnum.BRANCH
+
+    name = properties.String('name')
+    uid = properties.Optional(properties.UUID(), 'id')
+    created_at = properties.Optional(properties.Datetime(), 'created_at', serializable=False)
+    updated_at = properties.Optional(properties.Datetime(), 'updated_at', serializable=False)
+
+    project_id: Optional[UUID] = None
+
+    def __init__(self,
+                 name: str,
+                 *,
+                 session: Optional[Session] = None):
+        self.name: str = name
+        self.session: Session = session
+
+    def __str__(self):
+        return f'<Branch {self.name!r}>'
+
+    def _path(self):
+        return format_escaped_url('/branch/{branch_id}', branch_id=self.uid)
+
+    @property
+    def design_workflows(self) -> DesignWorkflowCollection:
+        """Return a resource representing all workflows contained within this branch."""
+        if getattr(self, 'project_id', None) is None:
+            raise AttributeError('Cannot initialize workflow without project reference!')
+        return DesignWorkflowCollection(project_id=self.project_id, branch_id=self.uid, session=self.session)
+
+
+class BranchCollection(Collection[Branch]):
```

Comment:
I left `delete` alone since it's currently present in the API docs, although I should note that when trying it against development I get a 500. And as archiving branches is still being discussed, I figure that should probably be a follow-on ticket.

Reply:
Go ahead and do a NotImplementedError for delete for now, we can follow up with an archiving method or a delete implementation depending on what we go with in the backend. Unless we expect delete branch to work (checking with andy)

Reply:
Confirmed with Andy that we expect to be able to delete an empty branch. I filed a ticket for us to more gracefully decline when the user requests we delete a non-empty branch: https://citrine.atlassian.net/browse/PLA-8755.

Also note, there are two issues which arise when deleting an empty branch. One is the Response object returned does not contain the status code due to how Session.delete_resource works: https://citrine.atlassian.net/browse/PLA-8764. The other is that it also doesn't contain the ID of the deleted branch due to the endpoing returning the wrong Content-Type: https://citrine.atlassian.net/browse/PLA-8765